cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -319,15 +319,19 @@ runTxBuildCmd
   -> TxMetadataJsonSchema
   -> [ScriptFile]
   -> [MetadataFile]
-  -> Maybe ProtocolParamsFile
+  -> Maybe (Deprecated ProtocolParamsFile)
   -> Maybe UpdateProposalFile
   -> TxBuildOutputOptions
   -> ExceptT ShelleyTxCmdError IO ()
 runTxBuildCmd
     socketPath (AnyCardanoEra cEra) consensusModeParams@(AnyConsensusModeParams cModeParams)
     nid mScriptValidity mOverrideWits txins readOnlyRefIns
     reqSigners txinsc mReturnColl mTotCollateral txouts changeAddr mValue mLowBound
-    mUpperBound certs wdrls metadataSchema scriptFiles metadataFiles mProtocolParamsFile mUpProp outputOptions = do
+    mUpperBound certs wdrls metadataSchema scriptFiles metadataFiles mDeprecatedProtocolParamsFile
+    mUpProp outputOptions = do
+  forM_ mDeprecatedProtocolParamsFile $ \_ ->
+    liftIO $ printWarning "'--protocol-params-file' for 'transaction build' is deprecated"
```

Comment:
Warn early rather than print warnings deep within the call hierarchy and after doing work.